rest_framework_serializer_extensions/serializers.py

```diff
@@ -409,6 +429,21 @@ def _get_expand_fields(self, standard_fields):
 
         return expanded_fields
 
+    def run_validation(self, data=empty):
+        """
+        After validation, convert writable <fieldname>_id fields to <fieldname>
+        """
+        validated_data = (
+            super(ExpandableFieldsMixin, self).run_validation(data=data)
+        )
+        for id_field_name in self._id_fields_to_translate:
+            if id_field_name in validated_data:
+                model_field_name = id_field_name[:-3]
+                validated_data[model_field_name] = (
+                    validated_data.pop(id_field_name)
+                )
```

Comment:
It looks like you'll get a problem here if you *also* expand the field normally during expansion. If the view set an `extensions_expand = {'x'}` attribute, then x will have a collision.

I think it's probably worth thinking over the situations where you might want to post expanded fields. Granted, it seems weird that you would want to update an attribute on a foreign relation from a single post, but as you don't *only* expand foreign relations, there may be legitimate use cases for doing so.